scripts/Diffraction/isis_powder/routines/focus.py

```diff
@@ -57,13 +60,52 @@ def _focus_one_ws(input_workspace, run_number, instrument, perform_vanadium_norm
 
     # Correct for absorption / multiple scattering if required
     if absorb:
-        input_workspace = instrument._apply_absorb_corrections(run_details=run_details, ws_to_correct=input_workspace)
+        # input_workspace = instrument._apply_absorb_corrections(run_details=run_details, ws_to_correct=input_workspace)
+
+        corrections = mantid.PaalmanPingsMonteCarloAbsorption(
+            InputWorkspace=input_workspace,
+            Shape='Cylinder',
+            BeamHeight=2.0,
+            BeamWidth=2.0,
+            Height=2.0,
+            SampleWidth=2.0,
+            SampleThickness=0.1,
+            SampleChemicalFormula='H2-O',
+            SampleDensity=1.0,
+            ContainerFrontThickness=0.02,
+            ContainerBackThickness=0.02,
+            ContainerChemicalFormula='V',
+            ContainerDensity=6.0,
+            CorrectionsWorkspace='corrections'
+        )
+        # ass_ws = corrections[0]
+        # assc_ws = corrections[1]
+        # acsc_ws = corrections[2]
+        # acc_ws = corrections[3]
+
+        input_workspace = mantid.ApplyPaalmanPingsCorrection(SampleWorkspace=input_workspace,  # ass_ws ??
+                                                             CorrectionsWorkspace=corrections,
+                                                             CanWorkspace=can_ws) # Where from?
```

Comment:
I think you should probably set the value of the CanWorkspace parameter to `summed_empty`. The `summed_empty` workspace is the sum of one or more runs done with an empty container or can. 
Also if you're doing this you don't need L54-55 where `subtract_summed_runs` is called. The Paalman Pings process subtracts the empty can measurement and does the absorption correction
One for when we get back after Christmas though I think!

Reply:
Review comments are still valid: I will deal with them separately: https://github.com/mantidproject/mantid/issues/33030